kernel/src/process/posix_thread/posix_thread_ext.rs

```diff
@@ -11,17 +11,12 @@ use crate::{
     fs::fs_resolver::{FsPath, FsResolver, AT_FDCWD},
     prelude::*,
     process::{process_vm::ProcessVm, program_loader::load_program_to_vm, Credentials, Process},
-    thread::{Thread, Tid},
+    thread::{Thread, ThreadExt, Tid},
 };
+
+/// An extension trait for some [`PosixThread`]-like types.
 pub trait PosixThreadExt {
```

Comment:
Do you expect other methods to be added to `ThreadExt` and `PosixThreadExt`. If not, then `ThreadExt` and `PosixThreadExt` should be named `AsThread` and `AsPosixThread`. The latter names are more accurate.

Reply:
No, I want to avoid adding more methods to these traits. Writing `as_thread` and then accessing the fields should be clearer.

So `AsThread` is definitely a good name. Done the renaming in the new commit, since the `PosixThreadExt` naming existed before this PR.